Black Clovedle/ViewController.swift

```diff
@@ -6,14 +6,710 @@
+        tableView.register(UINib(nibName: "Guess", bundle: nil), forCellReuseIdentifier: "GuessCell")
+        guessingTableView.isHidden = true
+        tableView.reloadData()
+        
+        if currentDate == UserDefaults.standard.string(forKey: "lastUpdatedDate") {
+            correctName = UserDefaults.standard.string(forKey: "winningCharacter")
+            performSegue(withIdentifier: "CharacterCorrectViewController", sender: nil)
+            
+            
+        }
+        
+        guessText.textContentType = .name
+        
+       
+    }
+    
+  
+    
+    override func viewDidAppear(_ animated: Bool) {
+        submit.tintColor = .black
+        print("viewdidappearran")
+        updateCurrentDate()
+        getRandomCharacter()
+        tableView.reloadData()
+        if currentDate == UserDefaults.standard.string(forKey: "lastUpdatedDate") {
+            correctName = UserDefaults.standard.string(forKey: "winningCharacter")
+            performSegue(withIdentifier: "CharacterCorrectViewController", sender: nil)
+        }
+        if isKeyPresentInUserDefaults(key: "lastDatePlayed") == true {
+            if UserDefaults.standard.object(forKey: "lastDatePlayed") as! String == currentDate {
+                correctName = UserDefaults.standard.string(forKey: "winningCharacter")
+                performSegue(withIdentifier: "CharacterCorrectViewController", sender: nil)
+                
+            }
+        } else{
+            
+        }
+    }
+    
+    
+    @objc func appDidBecomeActive() {
+        print("app did become active")
+        
+        updateCurrentDate()
+        if currentGameDate != currentDate {
+            print("did this run?")
+            clearGuessTable()
+            resetGame()
+            
+        } else {
+            resetGame()
+            getRandomCharacter()
+            loadGuessTable()
+            
+            
+            DispatchQueue.main.async{
+                for guess in self.guesses {
+                    self.rebuildTable(name: guess)
+                    
+                }
+                
+                self.tableView.reloadData()
+                
+            }
+        }
+       
+       
+        
+   
+        
+    }
+    
+//    
+//    func saveImagesToUserDefaults(images: [UIImage], key: String) {
+//        let imageDataArray = images.compactMap { $0.pngData() }
+//        UserDefaults.standard.set(imageDataArray, forKey: key)
+//    }
+//
+//    func loadImagesFromUserDefaults(key: String) -> [UIImage] {
+//        guard let imageDataArray = UserDefaults.standard.array(forKey: key) as? [Data] else {
+//            return []
+//        }
+//        return imageDataArray.compactMap { UIImage(data: $0) }
+//    }
+
+   
+    
+    @IBAction func Chevron(_ sender: Any) {
+        changeChevronState()
+    }
+    
+    
+    
+    func isKeyPresentInUserDefaults(key: String) -> Bool {
```

Comment:
If you don't have a server, UserDefaults is a great tool to keep track of things. With that said, it's often nice to have an added abstraction over it. Something like:

```
enum AppDefaults { 
  static var currentDate: String { ... }
  static func setCurrentDate(_ dateString: String) { ... }
  static func isKeyPresent(key: String) -> Bool { ... }
}
```

This specific function `isKeyPresentInUserDefaults` could technically be a free function, but it's nice to use the convention above to namespace things for readability.